libpod/runtime_img.go

```diff
@@ -299,11 +297,12 @@ func (r *Runtime) LoadImage(ctx context.Context, name, inputFile string, writer
 			return layout.NewReference(inputFile, name)
 		},
 		func() (types.ImageReference, error) {
+			iName := name // name may be ""
 			// prepend "localhost/" to support local image saved with this semantics
-			if !strings.Contains(name, "/") {
-				return layout.NewReference(inputFile, fmt.Sprintf("%s/%s", image.DefaultLocalRegistry, name))
+			if iName != "" && !strings.Contains(name, "/") {
+				iName = fmt.Sprintf("%s/%s", image.DefaultLocalRegistry, name)
 			}
-			return nil, nil
+			return layout.NewReference(inputFile, iName)
```

Comment:
(In both the `ociarchive` and `layout` code above:)
- AFAICS this is unrelated to #7337 , and probably belongs to a separate PR.
- If the separate `NewReference(inputFile, name)` attempt remains (as it should), this case should _only_ try prepending `DefaultLocalRegistry`. It’s a good point that in the `""` case that shouldn’t happen, but in that case these functions should return `nil, nil`, not try to load the archive the second time using the same reference that has already failed.